From 1a21c5913c52fc9812e0f18f88a3947f8f613208 Mon Sep 17 00:00:00 2001
From: repo sync -j8 --no-tags -qc   cd qsdk <quic_tganatra@quicinc.com>
Date: Tue, 23 Jul 2024 18:05:25 +0530
Subject: [PATCH] [tcrypt] AHASH changes for kernel

This patch has enabling AHASH_TYPE_2(multiple udpate request for ahash) test cases.

Change-Id: I6763ac4ad8f649e39728d74018694938f67ab8a0
Signed-off-by: <quic_tganatra@quicinc.com>
---
 crypto/Kconfig   | 6 ------
 crypto/testmgr.c | 6 ------
 2 files changed, 12 deletions(-)

diff --git a/crypto/Kconfig b/crypto/Kconfig
index 763cb6e7701c..65094a608583 100644
--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -248,12 +248,6 @@ config CRYPTO_NO_AES_XTS_ZERO_KEY_SUPPORT
 	help
 	  Disable support for zero key test cases
 
-config CRYPTO_DISABLE_AHASH_TYPE2_TESTS
-	bool "Disable AHASH type 2 test cases"
-	default y
-	help
-	  Disable AHASH type 2 tests
-
 config CRYPTO_DISABLE_AHASH_TYPE3_TESTS
 	bool "Disable AHASH type 3 test cases"
 	default y
diff --git a/crypto/testmgr.c b/crypto/testmgr.c
index 82a4a8c0fdaf..44135ecea75a 100644
--- a/crypto/testmgr.c
+++ b/crypto/testmgr.c
@@ -425,11 +425,6 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 		.key_offset = 1,
 		.key_offset_relative_to_alignmask = true,
 	},
-#ifndef CONFIG_CRYPTO_DISABLE_AHASH_TYPE2_TESTS
-	/*
-	 * Update in testmgr requires the result back whereas HW hides result from the user
-	 * TODO : Require support for ahash multiple update
-	 */
 	{
 		.name = "init+update+update+final two even splits",
 		.src_divs = {
@@ -441,7 +436,6 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 		},
 		.finalization_type = FINALIZATION_TYPE_FINAL,
 	},
-#endif
 	{
 		.name = "digest uneven misaligned splits, may sleep",
 		.req_flags = CRYPTO_TFM_REQ_MAY_SLEEP,
-- 
2.34.1

